atlas_densities/densities/inhibitory_neuron_densities_optimization.py

```diff
@@ -481,21 +481,42 @@ def _check_variables_consistency(
         AtlasDensitiesError if on the the following assumptions is violated:
         - if cell count estimate of a region is known with certainty for a given cell type,
         then the cell count of every descendant region is also known with certainty.
-        - a cell count estimate which is given for certain does not
+        - a neuron subtype count estimate which is given for certain is greater than
+        its total neuron count estimate counterpart.
     """
     cell_count_tolerance = 1e-2  # absolute tolerance to rule out round-off errors
+    # pylint: disable=too-many-nested-blocks
     for region_name, id_, id_set in zip(
         deltas.index, hierarchy_info.index, hierarchy_info["descendant_id_set"]
     ):
         for cell_type in cell_types:
             if np.isfinite(deltas.loc[region_name, cell_type]):
                 for desc_id in id_set:
                     if np.isnan(x_result.loc[desc_id, cell_type]):
-                        raise AtlasDensitiesError(
-                            f"Cell count estimate of region named '{region_name}' for cell type "
-                            f"{cell_type} was given for certain whereas the cell count of "
-                            f"descendant id {desc_id} is not certain."
-                        )
+                        if (deltas.loc[region_name, cell_type] in [0.0, np.inf]) and (
+                            (x_result.loc[id_, cell_type] == 0.0)
+                            or (np.isnan(x_result.loc[id_, cell_type]))
+                        ):
+                            # If the region's cell count value was set to 0 because the region does
```

Comment:
@cveraszto, if this "set to 0" was done by hardcoding `0.0`,  it would be safer to define a variable (say `zero_count = 0.0`) and reuse it here instead of hardcoding `0.0` again.
Same for the `0.0` in `x_result.loc[id_, cell_type] == 0.0`

Reply:
We can do a variable `zero_count`, but just so we both understand each other what I mean under `set to 0` is: `in _compute_initial_cell_counts()` we calculate volumes with `compute_region_volumes()` which will be the multiplicator for calculating densities (number of cells * volume = `nr of cells / mm^3` (in denominator). It's not hard coded per se (in my opinion) it's just calculating densities and for zero volumes it will be `0` since there can't be cells if space is `0`. (And this method of calculating density fails when we feed `np.nan`s to it, and here we take care of this issue: `np.nan`s will turn into `0.0`).  

Your suggestion is code safety and good practices related, right?

Reply:
Right, and as you said the zero value is not hard-coded, it comes from [this](https://github.com/BlueBrain/atlas-densities/blob/416b05e2b2937f16d8863269a78cf3c46a3dae88/atlas_densities/densities/utils.py#L543) `.sum()` if I am correct.
So nothing to change from my side, maybe just "was set to 0" with "is 0" in the comment.